### PR TITLE
Py3.command_output() signal fix

### PR DESCRIPTION
**Pull request by tobes**
In #664 an issue was identified where a command would sometimes get terminated by a linux signal (this is what the negative return codes are about).  I think this is something to do with an empty STDIN in that issue.  It may also be a python 3.6 related change.

Anyhow this fixes this which to me feels the correct approach.

### CODE REVIEW

**Comment by lasers** [comment]
This disables `xsel` module and I agree with this simple approach.

I'm able to `i3-msg restart` without facing any crash thanks to @tobes after several long hours of troubleshooting and finally resolves several strange issues with `xsel`.

**Comment by ultrabug** [comment]
I'm perplex here as I don't see how it can happen and even more why we should ignore it, it must be late for me but why is a linux killed command okay to forgive when a non 0 exit value is not ?

**Comment by tobes** [comment]
> I'm perplex here as I don't see how it can happen and even more why we should ignore it, it must be late for me but why is a linux killed command okay to forgive when a non 0 exit value is not ?

Personally I think it is a bug in python 3.6, as a release it seems to have made changes that I really don't understand.  I haven't yet had a chance to play with it but the fact that `dir(str)` no longer includes things like `__init__` scares me a little.

Anyhow the upshot is that for @lasers at least is getting errors using `subprocess.check_output()` is failing for the same reason.  py3.command_output()` is pretty much the same code as `subprocess.check_output()` anyhow.  As an example running `xsel` fails consistently for him even though the command returns the correct command output.  So imho it is a problem somewhere in python.

I'd like to understand it more too, but it has been spotted in several unrelated places. So it is a issue for some current users.

**Comment by lasers** [comment]
@ultrabug Hi there... Sir... _takes the hat off_

I tried xsel. It didn’t work for me. I told @tobes this. He told me to fix xsel. So I did that...

I fixed xsel. This. https://github.com/ultrabug/py3status/pull/664. The code, I believe, is ready. I believe @tobes too considered it ready.

However, I shared with @tobes my great sadness and disappointment in fixing xsel is still not curing me of my xsel sickness.

From there, @tobes, a great doctor, ordered me to go on an adventure with him to find a cure for my xsel sickness. He believes it to be some kind of exotic bug. I knew nothing of it.

Off we went... with our nets and insect repellent to an uninhabited swamp crawling with bugs everywhere. We continues to navigate through... Back and forward. Left and right. We kept looking for some kind of exotic bug... Slowly but surely. Soon enough, we found something.. and we trapped it.

We played with it. We make it scared. We push it. We make it sing. We make it spitting out goo. Finally, we tested how it would reacts if we were to change from `if retcode:` to `if retcode > 0:`. It stopped. The crashes. it stopped.

There, @tobes squashed the exotic baby bug... _figuratively speaking._

My hero. He saved me. He cured me of xsel sickness. We went back home.

Weeks later...

Wanting to eradicate the world of xsel sickness, @tobes publishes his medical journey on this pull request including everything he knew about xsel sickness including the surprising treatment. He explained about how the natives spoke of baby exotic bugs. It were known as @ultrabug in their language.

# *BOOM!* A loud earthquake boom.


Without warning, a huge @ultrabug appears out of the ground... filled with angry... demanded why we are doing this. He knew if @tobes got this pull request merged, the exotic bugs in his uninhabited swamp could get eradicated. This would mean no more @ultrabug.  

_"I'm perplexed here! Why do you do this?!? My baby bugs!"_ roared the exotic @ultrabug

We're doing this because of...

* **xsel: Fix sel** _Pull request_ https://github.com/ultrabug/py3status/pull/664
* **new module** _external.py. works xsel, xclip, etc for me_ https://github.com/ultrabug/py3status/pull/663

* **xsel bug** _xsel error with python3 when less than max_width_ https://github.com/ultrabug/py3status/issues/670
* **more xsel bug** _xsel: ERROR (CalledProcessorError)_ https://github.com/ultrabug/py3status/issues/619
* **somehow more xsel bug** _Spawning yet another py3status_ https://github.com/ultrabug/py3status/issues/634

**tldr:** something fuckery is going on and we don't know why.

**Comment by tobes** [comment]
> tldr: something fuckery is going on and we don't know why. idk.

Which is why we try to understand the fuckery __before__ making changes.  So on this @ultrabug is correct.

**Comment by lasers** [comment]
Unpublished modified `bluetooth` using helpers. // Sat Jan 28 22:48:32 CST 2017

`Exception: Command 'hcitool' returned non-zero exit status -15`
```
2017-01-28 22:33:16 WARNING Instance `bluetooth`, user method `bluetooth` failed (Exception) py3.py line 699.
2017-01-28 22:33:16 INFO Traceback
Exception: Command 'hcitool' returned non-zero exit status -15
  File "/home/chris/src/py3status/py3status/module.py", line 584, in run
    response = method()
  File "/home/chris/src/py3status/py3status/modules/bluetooth.py", line 44, in bluetooth
    full_text = self._get_bluetooth_name()
  File "/home/chris/src/py3status/py3status/modules/bluetooth.py", line 81, in _get_bluetooth_name
    name = self.py3.command_output('hcitool name %s' % mac).strip()
  File "/home/chris/src/py3status/py3status/py3.py", line 699, in command_output
    raise Exception(msg.format(cmd=command[0], error=retcode))

2017-01-28 22:33:16 ERROR py3status: Instance `bluetooth`, user method `bluetooth` failed (Exception) py3.py line 699. Please try to fix this and reload i3wm (Mod+Shift+R)

**Comment by tobes** [comment]
I've updated this so that we only ignore SIGTERM (-15) which is the issue that affected @lasers.  I still don't understand the issue but can confirm that it affects a VM running arch and happens with both python 2.7 and 3.6.  I'm assuming it is not hit commonly or we would have more reports.

@lasers can you see if this new patch helps with your issue.  (I'm assuming you still are hitting this quite often)

**Comment by lasers** [comment]
Before: CRASH... RED BAR... CRASH... RED BAR... CRASH... RED BAR!
After: No crash. `xsel` silently killed. Log reported `ignoring`. Smooth. Bliss. 
```
2017-02-01 07:14:39 INFO method dpms returned {'full_text': 'DPMS', 'color': '#00FF00', 'instance': '', 'name': 'dpms'}
2017-02-01 07:14:42 INFO lock cleared, exiting
2017-02-01 07:14:42 INFO Module `xsel`: Command `['xsel']` returned SIGTERM (ignoring)
2017-02-01 07:14:42 INFO method xsel returned {'full_text': '    fo..con}"\n', 'instance': '', 'name': 'xsel'}
2017-02-01 07:14:43 INFO ========
2017-02-01 07:14:43 INFO Starting py3status version 3.5rc0 python 3.6.0
2017-02-01 07:14:43 INFO git branch: ultrabug-command_output-signal-fix
2017-02-01 07:14:43 INFO git commit: 0858e11 Created from ultrabug/command_output-signal-fix
```

**Comment by tobes** [comment]
Please do not use `RED BAR... ` as a term it is confused.  What you see is a notification.  It happens that it is an i3-nagbar and it happens to be red.

just call it an error notification it'll help resolve issues.  red bar sounds like an i3bar thing which this is not :)

@ultrabug I think we should take this.  I've added a merge label, feel free to remove it.

**Comment by ultrabug** [comment]
Okay, thanks for the clarification, I agree with you guys that it should be merged.
